api/compaction_manager.cc

```diff
@@ -75,11 +75,11 @@ void set_compaction_manager(http_context& ctx, routes& r) {
     cm::get_pending_tasks_by_table.set(r, [&ctx] (std::unique_ptr<request> req) {
         return ctx.db.map_reduce0([&ctx](database& db) {
             std::unordered_map<std::pair<sstring, sstring>, uint64_t, utils::tuple_hash> tasks;
-            return do_for_each(db.get_column_families(), [&tasks](const std::pair<utils::UUID, seastar::lw_shared_ptr<table>>& i) {
+            return do_for_each(db.get_column_families(), [tasks = std::move(tasks)](const std::pair<utils::UUID, seastar::lw_shared_ptr<table>>& i) mutable {
                 table& cf = *i.second.get();
                 tasks[std::make_pair(cf.schema()->ks_name(), cf.schema()->cf_name())] = cf.get_compaction_strategy().estimated_pending_compactions(cf);
                 return make_ready_future<>();
-            }).then([&tasks] {
+            }).then([tasks = std::move(tasks)]() mutable {
```

Comment:
I may not be parsing the code correctly, but it looks to me you **move** tasks into the do_for_each lambda, and then move it, again, in the then - isn't this a bug?
Another thing you could do here is to just use a do_with() to hold the tasks instead of it being a local variable - then the &tasks would be fine.

Reply:
the problem with do_with is that you cannot return the parameter you hold

Reply:
I'll send a different version

Reply:
You can return the parameter, if you make a copy of it (you may need to explicitly define the type of the return from the lambda, using ->, so it doesn't return a pointer?). But I'm not familiar enough with the code to say what it is trying to do. I just thought the two moves were wrong.